### PR TITLE
feat(device_mqtt, device_rest): migrate device-list files for device-…

### LINKED ISSUE #72: Migrate Device MQTT and Device REST device files.
Theses file need to be converted to YAML with proper camelCase property names. Case for names (map keys) under `Protocols`  must stay as current casing.

https://github.com/intel-retail/rtsf-at-checkout-reference-design/blob/EdgeX-3.0/loss-detection-app/res/device-mqtt/devices/device-list.toml
https://github.com/intel-retail/rtsf-at-checkout-reference-design/blob/EdgeX-3.0/loss-detection-app/res/device-rest/devices/device-list.toml

### PR DESCRIPTION
**Pull request by seanohair22**
…mqtt and device-rest to EdgeX v3.0 yaml

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?

- Updates device-mqtt and device-rest device files to yaml for EdgeX v3.0

## Issue this PR will close

close: #**72**

closes #72 

## Test Instructions if applicable

- Run `make run-edgex run-portainer`
- Navigate to `localhost:9000` and check portainer logs of `device-rest` and `device-mqtt`
- Look for `level=INFO ts=2023-10-03T17:32:39.903218108Z app=device-mqtt source=devices.go:54 msg="Loading pre-defined devices from /res/devices(1 files found)"`
 and `level=INFO ts=2023-10-03T17:32:39.832931714Z app=device-rest source=devices.go:54 msg="Loading pre-defined devices from /res/devices(1 files found)"`
- Close with `make clean` and `docker compose -f loss-detection-app/docker-compose.portainer.yml down`

### CODE REVIEW

**Review by marcpfuller — APPROVED** [review]
LGTM

File: loss-detection-app/res/device-mqtt/devices/device-list.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+deviceList:
+  - name: pos-mqtt
+    profileName: pos
+    description: MQTT device for POS
+    labels:
+      - MQTT
+      - pos
+    protocols:
+      mqtt:
+        CommandTopic: EdgeXPosCommand
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-mqtt/devices/device-list.yaml:14]: should this be in a list format?

File: loss-detection-app/res/device-mqtt/devices/device-list.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  - name: scale-mqtt
+    profileName: scale
+    description: MQTT device for Scale
+    labels:
+      - MQTT
+      - scale
+    protocols:
+      mqtt:
+        CommandTopic: EdgeXScaleCommand
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-mqtt/devices/device-list.yaml:23]: ditto

File: loss-detection-app/res/device-mqtt/devices/device-list.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  - name: rfid-roi-mqtt
+    profileName: rfid-roi
+    description: MQTT device for RFID ROI events
+    labels:
+      - MQTT
+      - rfid
+    protocols:
+      mqtt:
+        CommandTopic: EdgeXRfidCommand
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-mqtt/devices/device-list.yaml:32]: ditto

File: loss-detection-app/res/device-mqtt/devices/device-list.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  - name: cv-roi-mqtt
+    profileName: cv-roi
+    description: MQTT device for CV ROI events
+    labels:
+      - MQTT
+      - cv
+      - ROI
+    protocols:
+      mqtt:
+        CommandTopic: EdgeXRoiCommand
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-mqtt/devices/device-list.yaml:42]: ditto

File: loss-detection-app/res/device-rest/devices/device-list.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+deviceList:
+  - name: pos-rest
+    profileName: pos
+    description: RESTful Device that sends in JSON POS data
+    labels:
+      - REST
+      - json
+      - pos
+    protocols:
+      other: {}
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-rest/devices/device-list.yaml:15]: list format?

File: loss-detection-app/res/device-rest/devices/device-list.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  - name: cv-roi-rest
+    profileName: cv-roi
+    description: RESTful device for CV ROI events
+    labels:
+      - REST
+      - cv
+      - ROI
+    protocols:
+      other: {}
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-rest/devices/device-list.yaml:24]: ditto

File: loss-detection-app/res/device-rest/devices/device-list.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  - name: rfid-roi-rest
+    profileName: rfid-roi
+    description: RESTful device for RFID ROI events
+    labels:
+      - REST
+      - RFID
+      - ROI
+    protocols:
+      other: {}
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-rest/devices/device-list.yaml:33]: ditto

**Review by lenny-intel — APPROVED** [review]
LGTM

File: loss-detection-app/res/device-rest/devices/device-list.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  - name: scale-rest
+    profileName: scale
+    description: RESTful device for Scale
+    labels:
+      - REST
+      - scale
+    protocols:
+      other: {}
```
Comment (jim-wang-intel) [inline loss-detection-app/res/device-rest/devices/device-list.yaml:41]: ditto

**Review by brian-intel — APPROVED** [review]

File: loss-detection-app/res/device-mqtt/devices/device-list.yaml (hunk 1 above)
Comment (presatish) [inline loss-detection-app/res/device-mqtt/devices/device-list.yaml:14]: Should all others start with caps as this one? I see in the toml file that all started with caps.

Comment (lenny-intel) [reply loss-detection-app/res/device-mqtt/devices/device-list.yaml:14]: Toml case didn't matter. Yaml is camelCase except for the Protocol map keys which are up to the Device Service. In this case they use lowe case for the protocol name `mqtt` and PascalCase for the protocol setting name `CommandTopic`

**Review by presatish — APPROVED** [review]
LGTM

**Pull request merged**